network/debinterfaces/bridge_test.go

```diff
@@ -0,0 +1,326 @@
+    map home,*,*,*                  home
+    map work,*,*,00:11:22:33:44:55  work-wireless
+    map work,*,*,01:12:23:34:45:50  work-static`
+	s.assertBridge(input, input[1:], c, map[string]string{"eth0": "br-eth0"})
+}
+
+func (s *BridgeSuite) TestBridgeBondMaster(c *gc.C) {
+	input := `
+auto ens5
+iface ens5 inet manual
+    bond-lacp_rate slow
+    mtu 1500
+    bond-master bond0
+    bond-xmit_hash_policy layer2
+    bond-mode active-backup
+    bond-miimon 100`
+	s.assertBridge(input, input[1:], c, map[string]string{"eth0": "br-eth0"})
+}
+
+func (s *BridgeSuite) TestBridgeNoIfacesDefinedFromFile(c *gc.C) {
+	stanzas, err := debinterfaces.ParseSource("testdata/ifupdown-examples", nil, s.expander)
+	c.Assert(err, gc.IsNil)
+	input := format(stanzas)
+	s.assertBridge(input, input, c, map[string]string{"non-existent-interface": "non-existent-bridge"})
+}
+
+func (s *BridgeSuite) TestBridgeAlias(c *gc.C) {
+	input := `
+auto eth0
+iface eth0 inet static
+    gateway 10.14.0.1
+    address 10.14.0.102/24
+
+auto eth0:1
+iface eth0:1 inet static
+    address 1.2.3.5`
+
+	expected := `
+auto eth0
+iface eth0 inet manual
+
+auto br-eth0:1
+iface br-eth0:1 inet static
+    address 1.2.3.5
+
+auto br-eth0
+iface br-eth0 inet static
+    gateway 10.14.0.1
+    address 10.14.0.102/24
+    bridge_ports eth0`
+	s.assertBridge(input, expected[1:], c, map[string]string{"eth0": "br-eth0", "eth0:1": "br-eth0:1"})
+}
+
+func (s *BridgeSuite) TestBridgeMultipleInterfaces(c *gc.C) {
+	input := `
+auto enp1s0f3
+iface enp1s0f3 inet static
+  address 192.168.1.64/24
+  gateway 192.168.1.254
+  dns-nameservers 192.168.1.254
+  dns-search home
+
+iface enp1s0f3 inet6 dhcp`
+
+	expected := `
+auto enp1s0f3
+iface enp1s0f3 inet manual
+
+iface enp1s0f3 inet6 manual
+
+auto br-enp1s0f3
+iface br-enp1s0f3 inet static
+    address 192.168.1.64/24
+    gateway 192.168.1.254
+    dns-nameservers 192.168.1.254
+    dns-search home
+    bridge_ports enp1s0f3
+
+iface br-enp1s0f3 inet6 dhcp
+    bridge_ports enp1s0f3`
+	s.assertBridge(input, expected[1:], c, map[string]string{"enp1s0f3": "br-enp1s0f3"})
+}
+
+func (s *BridgeSuite) TestSourceStanzaWithRelativeFilenames(c *gc.C) {
+	stanzas, err := debinterfaces.Parse("testdata/TestInputSourceStanza/interfaces")
+	c.Assert(err, gc.IsNil)
+	c.Assert(stanzas, gc.HasLen, 3)
+	bridged := debinterfaces.Bridge(stanzas, map[string]string{"eth0": "br-eth0"})
+
+	expected := `
+auto lo
+iface lo inet loopback
+
+auto eth0
+iface eth0 inet manual
+
+auto eth1
+iface eth1 inet static
+    address 192.168.1.64
+    dns-nameservers 192.168.1.254
+
+auto eth2
+iface eth2 inet manual
+
+auto br-eth0
+iface br-eth0 inet dhcp
+    bridge_ports eth0`
+
+	c.Assert(debinterfaces.FormatStanzas(debinterfaces.FlattenStanzas(bridged), 4), gc.Equals, expected[1:])
+}
+
+func (s *BridgeSuite) TestSourceDirectoryStanzaWithRelativeFilenames(c *gc.C) {
+	stanzas, err := debinterfaces.Parse("testdata/TestInputSourceDirectoryStanza/interfaces")
```

Comment:
source-directory doesn't do globbing. It implicitly reads all files in `interfaces`.

`       Similarly,  "source-directory" keyword is used to source multiple files
       at once, without specifying them individually  or  using  shell  globs.
       Additionally,  when  "source-directory" is used, names of the files are
       checked to match the following regular expression: ^[a-zA-Z0-9_-]+$. In
       other words, the names must consist entirely of ASCII upper- and lower-
       case letters, ASCII digits, ASCII underscores, and ASCII minus-hyphens.
       In the directory path, shell wildcards may be used as well.`